rasa_sdk/interfaces.py

```diff
@@ -271,6 +272,35 @@ def add_slots(self, slots: List[EventType]) -> None:
             self.slots[event["name"]] = event["value"]
             self.events.append(event)
 
+    def get_intent_of_latest_message(
+        self, skip_fallback_intent: bool = True
+    ) -> Optional[Text]:
+        """Retrieves the intent the last user message.
+
+        Args:
+            skip_fallback_intent: Optionally skip the nlu_fallback intent
+                and return the next.
+        """
+        latest_message = self.latest_message
+        if not latest_message:
+            return None
+
+        intent_ranking = latest_message.get("intent_ranking")
```

Comment:
optional: pure style preference, but we could also do something like

```python
        intent_ranking = latest_message.get("intent_ranking", [])

        return next(
            (
                intent["name"]
                for intent in intent_ranking
                if not skip_fallback_intent
                or intent["name"] != NLU_FALLBACK_INTENT_NAME
            ),
            None,
        )
```

Reply:
I would tend to avoid using an iterator/generator if there is only max 2 elements that will ever be accessed.

Reply:
👍 because it's slower?

Reply:
No, I don't think it is slower in any meaningful way, it's a generator so it will not iterate the whole list (as it would if it was list comprehension).

I just think it adds a cognitive burden as the reader tends to try and understand the whole potential generator output, even if they know it's only the first element then get's retrieved.

Plus it's just a rule of thumb I use, that if there's less than three things, don't use a loop or list.

Although this is all quite subjective - so happy to have my mind changed!

Reply:
Thanks for the detailed explanation! I just wanted to understanding it a bit more. No strong opinion on it from my side 👍

Reply:
Always happy to discuss stuff like this :)